src/test/ui/generics/generic-impl-less-params-with-defaults.stderr

```diff
@@ -1,19 +1,19 @@
-error[E0107]: this struct takes at least 2 type arguments but only 1 type argument was supplied
+error[E0107]: this struct takes at least 2 generic arguments but 1 generic argument was supplied
   --> $DIR/generic-impl-less-params-with-defaults.rs:11:5
    |
 LL |     Foo::<isize>::new();
-   |     ^^^   ----- supplied 1 type argument
+   |     ^^^   ----- supplied 1 generic argument
    |     |
-   |     expected at least 2 type arguments
+   |     expected at least  2 generic arguments
    |
-note: struct defined here, with at least 2 type parameters: `A`, `B`
+note: struct defined here, with at least  2 generic parameters : `A`, `B`
   --> $DIR/generic-impl-less-params-with-defaults.rs:3:8
    |
 LL | struct Foo<A, B, C = (A, B)>(
    |        ^^^ -  -
-help: add missing type argument
+help: add missing generic argument
    |
-LL |     Foo::<isize, B>::new();
+LL |     Foo::<isize, A>::new();
```

Comment:
This should still be using `B`, right?

Reply:
Yes, probably. Although there were scenarios in which the previous implementation suggested the same parameter twice, which was the reason for why I settled with the current approach. I agree that this is strictly speaking  not correct, but then again implementing the suggestion as is, is also wrong since you don't use type parameters as generic arguments. I don't think this makes much difference, the user will probably always get what's going wrong here.